### PR TITLE
fix: webvpn login stuck on exception

### LINKED ISSUE #455: [BUG] 一旦 UIS 登录流程抛出异常，WebVPN 将再也无法使用
**描述 Bug**
`loginWebvpn` 方法没有捕获任何来自 `UISLoginTool.loginUIS` 的异常：

https://github.com/DanXi-Dev/DanXi/blob/4226b4a6a8c57c956872e1607bbf4be16ad8870a/lib/util/webvpn_proxy.dart#L123-L129

因此，一旦 `UISLoginTool.loginUIS` 抛出异常，在 `await` 时就会因抛出同步异常而中断函数，后续的 `loginSession = null;` 永远不可能被执行，故将导致 WebVPN Proxy 卡死在错误状态中，无论如何发送新请求都不会恢复。

cc @HydrogenC

### PR DESCRIPTION
**Pull request by HydrogenC**
Resolves #455.

### CODE REVIEW

File: lib/util/webvpn_proxy.dart [hunk 1]
```diff
@@ -118,16 +118,18 @@ class WebvpnProxy {
...
+        IndependentCookieJar workJar = IndependentCookieJar();
+        loginSession =
+            UISLoginTool.loginUIS(dio, WEBVPN_LOGIN_URL, workJar, _personInfo);
+        await loginSession;
+        // Clone from temp jar to our dedicated webvpn jar
+        webvpnCookieJar.cloneFrom(workJar);
+        isLoggedIn = true;
+      } finally {
+        loginSession = null;
+      }
```
Comment (w568w) [inline lib/util/webvpn_proxy.dart:132]: It will simply clean up and propagate any thrown exception. It this behavior expected?

Comment (HydrogenC) [reply lib/util/webvpn_proxy.dart]: Oh, I cut the `catch` and forgot to paste.

Comment (w568w) [reply lib/util/webvpn_proxy.dart]: > Oh, I cut the `catch` and forgot to paste.

In fact, I didn't think you made a mistake at the beginning; I just wanted to confirm if this is what you intended. Actually, NOT ignoring the error feels more user-friendly...

**Comment by HydrogenC** [comment]
Yeah, propagating the exception to be displayed on widgets will be a great option.

**Review by w568w — APPROVED** [review]
LGTM.

**Pull request merged**